markdown/developers/test_data_guidelines.md

```diff
@@ -0,0 +1,112 @@
+
+### Adding new test data
+
+1. Always ask on the [#modules](https://nfcore.slack.com/channels/modules) channel on the [nf-core slack](https://nf-co.re/join/slack) before adding test data.
+
+2. Add any new test data from a pull request from a personal fork of the nf-core/test-dataset repository.
+
+3. New bioinformatic test data files should be generated from the existing collections of test data as far as possible. See field-specific guidance [below](#field-specific-guidance).
+
+   For example, if you were adding new genomic test data, and needed to create a specific type of genotyping file format for a particular tool - you should use the SARS-CoV2 or _Homo sapiens_ BAM files as input when generating the file you will need to use your module tests. The resulting genotyping file should then be stored alongside the BAM file, or in the corresponding file format type directory.
+
+4. In the worst case scenario where you cannot (re)use the exisiting data in the field-specific collections, the `delete_me/` folder may be used.
+
+   - We discourage the use of this directory as far as possible.
+   - If you do use this directory, we ask you to replace anything added to this directory with proper test data based on the existing dataset as soon as possible.
+
+5. For non-bioinformatic specific files (like simple text files, or tables), you can place these in the `generic/` directory.
+
+6. Files must be small (less than the GitHub file-size limit as a maximum size), and can be aggressively subsampled if necessary.
+
+7. Test data must be publically available and have licenses that allow public reuse.
+
+8. Files should generally be organised in directories based on their file extension.
+
+9. Any test data files should be named based on the upstream file name and with a corresponding extension. For example, if you used `genome.fasta` as the upstream file, your output file should be called `genome.<new_extension>`.
+
+10. You must update the README file under the `Data Description` section of the modules-branch of test-datasets repostiroy to describe what the file is, and how it was generated (except for files added to `delete_me/`).
+
+11. The test data pull request requires a review to be merged.
+
+12. Once you've had your pull request merged in into `nf-core/test-datasets`, make another PR into [nf-core/modules](https://github.com/nf-core/modules), to add your file as a new entry to the [`modules/tests/config/test_data.config`](https://github.com/nf-core/modules/blob/master/tests/config/test_data.config).
+
+    - This only applies field-specific collections, `delete_me/` files are exclued from this, and raw github URLs should be used in tests.
+    - They 'key' for each URL should follow the style of the full file name with extensions but with underscores rather than full-stops e.g. `genome.fa.gz` would become `genome_fa_gz`.
```

Comment:
```suggestion
    - The 'key' for each URL should follow the style of the full file name with extensions but with underscores rather than full-stops e.g. `genome.fa.gz` would become `genome_fa_gz`.
```